src/graph_export/graph_export.cpp

```diff
@@ -0,0 +1,172 @@
+//***************************************************************************
+// Copyright 2025 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "graph_export.hpp"
+
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <utility>
+
+#pragma warning(push)
+#pragma warning(disable : 6313)
+#include <rapidjson/document.h>
+#include <rapidjson/stringbuffer.h>
+#include <rapidjson/writer.h>
+#pragma warning(pop)
+
+#include "../capi_frontend/server_settings.hpp"
+#include "../config.hpp"
+#include "../filesystem.hpp"
+#include "../logging.hpp"
+#include "../status.hpp"
+#include "../stringutils.hpp"
+
+namespace ovms {
+
+static std::string createTextGenerationGraphTemplate(const std::string& pipelineType, const std::string& modelPath, const std::string& maxNumSeqs, const std::string& targetDevice,
```

Comment:
Use GraphSettingsImpl instead of multiple parameters - If we add one we have to change multiple changes in code otherwise.